addons/purchase_edi_ubl_bis3/utils.py

```diff
@@ -0,0 +1,178 @@
+    'uom.product_uom_km': 'KMT',
+    'uom.product_uom_cm': 'CMT',
+    'uom.product_uom_litre': 'LTR',
+    'uom.product_uom_cubic_meter': 'MTQ',
+    'uom.product_uom_lb': 'LBR',
+    'uom.product_uom_oz': 'ONZ',
+    'uom.product_uom_inch': 'INH',
+    'uom.product_uom_foot': 'FOT',
+    'uom.product_uom_mile': 'SMI',
+    'uom.product_uom_floz': 'OZA',
+    'uom.product_uom_qt': 'QT',
+    'uom.product_uom_gal': 'GLL',
+    'uom.product_uom_cubic_inch': 'INQ',
+    'uom.product_uom_cubic_foot': 'FTQ',
+}
+
+# -------------------------------------------------------------------------
+# ELECTRONIC ADDRESS SCHEME (EAS), see https://docs.peppol.eu/poacc/billing/3.0/codelist/eas/
+# -------------------------------------------------------------------------
+EAS_MAPPING = {
+    'AD': {'9922': 'vat'},
+    'AL': {'9923': 'vat'},
+    'AT': {'9915': 'vat'},
+    'AU': {'0151': 'vat'},
+    'BA': {'9924': 'vat'},
+    'BE': {'0208': 'company_registry'},
+    'BG': {'9926': 'vat'},
+    'CH': {'9927': 'vat'},
+    'CY': {'9928': 'vat'},
+    'CZ': {'9929': 'vat'},
+    'DE': {'9930': 'vat'},
+    'DK': {'0184': 'company_registry', '0198': 'vat'},
+    'EE': {'9931': 'vat'},
+    'ES': {'9920': 'vat'},
+    'FI': {'0213': 'vat'},
+    'FR': {'0009': 'siret', '9957': 'vat'},
+    'SG': {'0195': 'l10n_sg_unique_entity_number'},
+    'GB': {'9932': 'vat'},
+    'GR': {'9933': 'vat'},
+    'HR': {'9934': 'vat'},
+    'HU': {'9910': 'vat'},
+    'IE': {'9935': 'vat'},
+    'IS': {'0196': 'vat'},
+    'IT': {'0211': 'vat', '0210': 'l10n_it_codice_fiscale'},
+    'JP': {'0221': 'vat'},
+    'LI': {'9936': 'vat'},
+    'LT': {'9937': 'vat'},
+    'LU': {'9938': 'vat'},
+    'LV': {'9939': 'vat'},
+    'MC': {'9940': 'vat'},
+    'ME': {'9941': 'vat'},
+    'MK': {'9942': 'vat'},
+    'MT': {'9943': 'vat'},
+    # Do not add the vat for NL, since: "[NL-R-003] For suppliers in the Netherlands, the legal entity identifier
+    # MUST be either a KVK or OIN number (schemeID 0106 or 0190)" in the Bis 3 rules (in PartyLegalEntity/CompanyID).
+    'NL': {'0106': None, '0190': None},
+    'NO': {'0192': 'l10n_no_bronnoysund_number'},
+    'NZ': {'0088': 'company_registry'},
+    'PL': {'9945': 'vat'},
+    'PT': {'9946': 'vat'},
+    'RO': {'9947': 'vat'},
+    'RS': {'9948': 'vat'},
+    'SE': {'0007': 'vat'},
+    'SI': {'9949': 'vat'},
+    'SK': {'9950': 'vat'},
+    'SM': {'9951': 'vat'},
+    'TR': {'9952': 'vat'},
+    'VA': {'9953': 'vat'},
+}
+
+# -------------------------------------------------------------------------
+# HELPERS
+# -------------------------------------------------------------------------
+
+
+def format_float(amount, precision_digits):
+    if amount is None:
+        return None
+    return float_repr(float_round(amount, precision_digits), precision_digits)
+
+
+def get_currency_decimal_places(currency_id):
+    # Allows other documents to easily override in case there is a flat max precision number
+    return currency_id.decimal_places
```

Comment:
Maybe we could depend on account_edi_ubl_cii for this builder module and use their tooling or the works they already did. purchase already depend on account